### PR TITLE
Speed up in constructing network

### PR DESCRIPTION
**Pull request by jisungyoon**
Hi! I am Jisung Yoon from POSTECH. 
Element-centric sim is awesome, and it works very well on overlapping community detection!

In this PR, I change some codes for constructing network (project the bipartite network)

In the previous code, you use the full matrix in calculating dot products and constructing networks.
It works well on small samples, but it takes an extremely long time in large samples.

I put some test notebook under the test codes, and I even cannot get the network because of the memory errors (It consumes more than 300 GB)

In the previous trial, it takes almost 1 hour and 30 minutes to calculate the network with using full cores (64 cores). But, with a new implementation, it takes only 2 minutes with small number of cores.

Please take a look, let me know what you think:) @ajgates42  @yy 

Thanks very much!

### CODE REVIEW

File: clusim/clusimelement.py [hunk 1]
```diff
@@ -267,7 +267,7 @@ def ppr_partition(clustering, alpha=0.9, relabeled_elements=None):
 
 
 def make_cielg(clustering, r=1.0, rescale_path_type='max',
-                relabeled_elements=None):
+                relabeled_elements=None, approach='previous'):
```
Comment (yy) [inline clusim/clusimelement.py]: I think the function body contains only the new approach? I think `sparse="true"` may be a better name?

**Review by yy — COMMENTED** [review]
Awesome! Just a small variable name suggestion.

File: clusim/clusimelement.py (hunk 1 above)
Comment (ajgates42) [reply clusim/clusimelement.py]: Yes, I'm always a big fan of keeping it simple, and sparse!

**Comment by ajgates42** [comment]
Hi @jisungyoon,
This looks great.  Thanks for the improvement!

File: clusim/clusimelement.py (hunk 1 above)
Comment (yy) [reply clusim/clusimelement.py]: Or, we can create another function `make_cielg_dense` or something.

**Comment by jisungyoon** [comment]
@yy Yeah, I just replace the codes, because it is always fast and return exactly the same results.
Do we need to keep the legacy code? @ajgates42

**Comment by yy** [comment]
Ok, I've removed the option. Now the thing is that the notebook will not run. Can we just keep the benchmark results here as a record but not the notebook?

**Comment by jisungyoon** [comment]
Screen Shot 2020-05-14 at 1 59 12 PM
Screen Shot 2020-05-14 at 1 59 19 PM
Screen Shot 2020-05-14 at 1 59 25 PM

**Comment by jisungyoon** [comment]
For archiving (legacy code)  
  proj1 = bipartite_adj / bipartite_adj.sum(axis=1)
  proj2 = bipartite_adj / bipartite_adj.sum(axis=0)
  projected_adj = proj1.dot(proj2.T)
  cielg = igraph.Graph.Weighted_Adjacency(projected_adj.tolist(),
                                             mode=igraph.ADJ_DIRECTED,
                                             attr="weight", loops=True)

**Review by yy — APPROVED** [review]

**Comment by yy** [comment]
@ajgates42 merge?

**Comment by ajgates42** [comment]
Nice, thanks @jisungyoon !

**Pull request merged**